app/src/main/java/net/programmierecke/radiodroid2/alarm/AlarmReceiver.java

```diff
@@ -215,6 +217,83 @@ protected void onPostExecute(String result) {
         }.execute();
     }
 
+    private void setAlarmVolume(float volume) {
+        try {
+            if (BuildConfig.DEBUG) { Log.d(TAG, "Setting volume to "+volume); }
+            itsPlayerService.SetVolume(volume);
+        } catch (Exception e) {
+            Log.e(TAG, "Couldn't set volume "+e);
+        }
+    }
+
+    private void graduallyIncreaseAlarmVolume(Context context) {
+        SharedPreferences sharedPref = PreferenceManager.getDefaultSharedPreferences(context);
+        int slowWakeMillis = sharedPref.getInt("gradually_increase_volume", 0) * 10000;
```

Comment:
There is 1000 ms in a second.

Reply:
And maybe name it something like `gradually_increase_volume_seconds` if that's what it means